fuse_core/test/test_throttled_callback.cpp

```diff
@@ -110,11 +117,29 @@ class PointSensorModel
    * @param[in] throttle_period The throttle period duration in seconds
    */
   explicit PointSensorModel(const rclcpp::Duration& throttle_period)
-    : throttled_callback_(std::bind(&PointSensorModel::keepCallback, this, std::placeholders::_1),
-                          std::bind(&PointSensorModel::dropCallback, this, std::placeholders::_1), throttle_period)
+    : Node("point_sensor_model_node")
+    , throttled_callback_(
+        std::bind(&PointSensorModel::keepCallback, this, std::placeholders::_1),
+        std::bind(&PointSensorModel::dropCallback, this, std::placeholders::_1),
+        throttle_period
+      )
   {
-    subscriber_ = node_handle_.subscribe<geometry_msgs::Point>(
-        "point", 10, &PointThrottledCallback::callback, &throttled_callback_);
+    subscription_ = this->create_subscription<geometry_msgs::msg::Point>(
+      "point", 10,
+      [&](const geometry_msgs::msg::Point::ConstSharedPtr& msg) {
+        this->throttled_callback_.callback(msg);
```

Comment:
nit, I'd recommend calling [via `operator()`](https://github.com/locusrobotics/fuse/blob/9f9395e5146e8bf0bbeedd47dc953ec974ab4187/fuse_core/include/fuse_core/throttled_callback.h#L194-L197) since that's what the test [is using in ROS 1](https://github.com/locusrobotics/fuse/blob/0461f779f9ac4defca6ecfd102a675add92bb3a9/fuse_core/test/test_throttled_callback.cpp#L116).

Reply:
I had to use a lambda because ROS 2's subscribe function is missing a signature for conveniently calling methods of objects. (Bind doesn't work nicely here, and there's a whole rabbit hole of arcane wizardry used to get it working.)

ROS 1 [has such a function signature](http://docs.ros.org/en/indigo/api/roscpp/html/classros_1_1NodeHandle.html#a317fe4c05919e0bf3fb5162ccb2f7c28)

With the rabbit hole encompassing:
- [the original call](https://github.com/strawlab/ros_comm/blob/master/clients/cpp/roscpp/include/ros/node_handle.h#L383)
- [initByFullCallbackType](https://github.com/strawlab/ros_comm/blob/6f7ea2feeb3c890699518cb6eb3d33faa15c5306/clients/cpp/roscpp/include/ros/subscribe_options.h#L82)

and inner factory functions.

I'm not too familiar with how ROS 1 does it to replicate it here (or upstream it for that matter..

Reply:
<details>

```
/home/methyldragon/Desktop/Projects/2022_10__locus_fuse/fuse_ws/src/fuse/fuse_core/test/test_throttled_callback.cpp: In constructor ‘PointSensorModel::PointSensorModel(const rclcpp::Duration&)’:
/home/methyldragon/Desktop/Projects/2022_10__locus_fuse/fuse_ws/src/fuse/fuse_core/test/test_throttled_callback.cpp:128:29: error: no matching function for call to ‘bind(<unresolved overloaded function type>, PointSensorModel::PointThrottledCallback*, const std::_Placeholder<1>&)’
  128 |       "point", 10, std::bind(&PointThrottledCallback::callback, &throttled_callback_, std::placeholders::_1));
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/methyldragon/Desktop/Projects/2022_10__locus_fuse/fuse_ws/src/fuse/fuse_core/include/fuse_core/throttled_callback.h:37,
                 from /home/methyldragon/Desktop/Projects/2022_10__locus_fuse/fuse_ws/src/fuse/fuse_core/test/test_throttled_callback.cpp:34:
/usr/include/c++/11/functional:789:5: note: candidate: ‘template<class _Func, class ... _BoundArgs> typename std::_Bind_helper<std::__is_socketlike<_Func>::value, _Func, _BoundArgs ...>::type std::bind(_Func&&, _BoundArgs&& ...)’
  789 |     bind(_Func&& __f, _BoundArgs&&... __args)
      |     ^~~~
/usr/include/c++/11/functional:789:5: note:   template argument deduction/substitution failed:
/home/methyldragon/Desktop/Projects/2022_10__locus_fuse/fuse_ws/src/fuse/fuse_core/test/test_throttled_callback.cpp:128:29: note:   couldn’t deduce template parameter ‘_Func’
  128 |       "point", 10, std::bind(&PointThrottledCallback::callback, &throttled_callback_, std::placeholders::_1));
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/methyldragon/Desktop/Projects/2022_10__locus_fuse/fuse_ws/src/fuse/fuse_core/include/fuse_core/throttled_callback.h:37,
                 from /home/methyldragon/Desktop/Projects/2022_10__locus_fuse/fuse_ws/src/fuse/fuse_core/test/test_throttled_callback.cpp:34:
/usr/include/c++/11/functional:813:5: note: candidate: ‘template<class _Result, class _Func, class ... _BoundArgs> typename std::_Bindres_helper<_Result, _Func, _BoundArgs>::type std::bind(_Func&&, _BoundArgs&& ...)’
  813 |     bind(_Func&& __f, _BoundArgs&&... __args)
      |     ^~~~
/usr/include/c++/11/functional:813:5: note:   template argument deduction/substitution failed:
/home/methyldragon/Desktop/Projects/2022_10__locus_fuse/fuse_ws/src/fuse/fuse_core/test/test_throttled_callback.cpp:128:29: note:   couldn’t deduce template parameter ‘_Result’
  128 |       "point", 10, std::bind(&PointThrottledCallback::callback, &throttled_callback_, std::placeholders::_1));
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gmake[2]: *** [test/CMakeFiles/test_throttled_callback.dir/build.make:76: test/CMakeFiles/test_throttled_callback.dir/test_throttled_callback.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:678: test/CMakeFiles/test_throttled_callback.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
```
</details>

Reply:
Edit: Seems like looking at it with fresh eyes worked

https://github.com/locusrobotics/fuse/compare/d126a1e1b67585f04ae2727fde188f2ed5cd44e9..536743131a2705e900eec50eee3c8fa15fc5dd1e